src/allocation_engine/README.md

```diff
@@ -67,6 +67,53 @@ they will be merged in a single node.
 
 ![Node Freeing example](/images/after_free.png)
 
+## Address aligning
+
+Memory slots allocated using address allocator can have the start aligned to a
+specified value. We offer the possibility to align the starting address either to
+the next boundary or the previous one.
+
+If the allocation policy used is `AllocPolicy::FirstMatch` all memory slots will
+start at an address aligned to the first multiple of specified alignment value
+that is greater or equal to the candidate address.
+
+Example:
+
+```rust
+initial_address              = 0b0000000000000000000000000000000001000000000000000000000000000001
+alignment                    = 0b0000000000000000000000000000000000000000000000000001000000000000
+initial_addr + alignment - 1 = 0b0000000000000000000000000000000001000000000000000001000000000000
+!(alignment - 1)             = 0b1111111111111111111111111111111111111111111111111111000000000000
+
+The aligned address will be the result of bitwise AND between `initial address + alignment - 1` and
+bitwise NOT of `alignment - 1`.
+
+0b0000000000000000000000000000000001000000000000000001000000000000(0x0000000040001000)&
+0b1111111111111111111111111111111111111111111111111111000000000000(0xFFFFFFFFFFFFF000)
+-------------------------------------------------------------------
+0b0000000000000000000000000000000001000000000000000001000000000000(0x0000000040001000)
+```
+
+If the allocation policy used is `AllocPolicy::LastMatch` all memory slots will
+start at an address aligned to the first multiple of specified alignment value
+that is lower or equal to the candidate address.
+
+Example:
+
+```rust
```

Comment:
This does not seem right. The code block is marked as rust, but that would not compile.